File: gaia-sdk-python/gaia_sdk/graphql/response/type/UpdatedTenantImpulse.py

from gaia_sdk.graphql.response.type.Tenant import Tenant

Uuid = str
String = str
ISO8601 = str
Struct = dict
Float = float
Int = int
Boolean = bool


class UpdatedTenantImpulse:
    """
    Impulse which indicates the result of a update Tenant impulse
    """
    dictionary: dict

    def __init__(self, dictionary: dict):
        self.dictionary = dictionary

    def __eq__(self, other):
        if type(other) is type(self):
            return self.dictionary == other.dictionary
        return False

    def __repr__(self):
        return {'dictionary': self.dictionary}

    @property
    def id(self) -> Uuid:
        return Uuid(self.dictionary.get("id"))
    """
    the Tenant instance
    """
    @property
    def data(self) -> Tenant:
        return Tenant(self.dictionary.get("data"))

File: gaia-sdk-javascript/src/graphql/response/type/UpdatedTenantImpulse.ts

import {Tenant} from "./Tenant";

import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {Order} from "../../request/enumeration/Order";
import {OrderByField} from "../../request/enumeration/OrderByField";
import {EdgeOrderByField} from "../../request/enumeration/EdgeOrderByField";

/**
* Impulse which indicates the result of a update Tenant impulse
*/
export interface UpdatedTenantImpulse {
    id?:Uuid, 
    /**
    * the Tenant instance
    */
    data?:Tenant
}